### ISSUE TITLE
UnboundLocalError: local variable 'request_id' referenced before assignment

### ISSUE THREAD

**Issue by bkb3** [flask-dashboard/Flask-MonitoringDashboard#245]
**Describe the bug**
Version 2.1.4 under RHEL 8, and python 3.6.8 gives the following error:

```
[Mon Jul 29 13:45:27.532457 2019] [wsgi:error] [pid 17797:tid 139631104370432] Exception in thread Thread-2:
[Mon Jul 29 13:45:27.532493 2019] [wsgi:error] [pid 17797:tid 139631104370432] Traceback (most recent call last):
[Mon Jul 29 13:45:27.532496 2019] [wsgi:error] [pid 17797:tid 139631104370432]   File "/usr/lib64/python3.6/threading.py", line 916, in _bootstrap_inner
[Mon Jul 29 13:45:27.532498 2019] [wsgi:error] [pid 17797:tid 139631104370432]     self.run()
[Mon Jul 29 13:45:27.532501 2019] [wsgi:error] [pid 17797:tid 139631104370432]   File "/usr/local/lib/python3.6/site-packages/flask_monitoringdashboard/core/profiler/stacktraceProfiler.py", line 77, in run
[Mon Jul 29 13:45:27.532503 2019] [wsgi:error] [pid 17797:tid 139631104370432]     self._on_thread_stopped()
[Mon Jul 29 13:45:27.532506 2019] [wsgi:error] [pid 17797:tid 139631104370432]   File "/usr/local/lib/python3.6/site-packages/flask_monitoringdashboard/core/profiler/stacktraceProfiler.py", line 93, in _on_thread_stopped
[Mon Jul 29 13:45:27.532508 2019] [wsgi:error] [pid 17797:tid 139631104370432]     self._outlier_profiler.add_outlier(request_id)
[Mon Jul 29 13:45:27.532511 2019] [wsgi:error] [pid 17797:tid 139631104370432] UnboundLocalError: local variable 'request_id' referenced before assignment
[Mon Jul 29 13:45:27.532517 2019] [wsgi:error] [pid 17797:tid 139631104370432] 
[Mon Jul 29 13:45:39.941647 2019] [wsgi:error] [pid 17797:tid 139631104370432] Exception in thread Thread-5:
[Mon Jul 29 13:45:39.941690 2019] [wsgi:error] [pid 17797:tid 139631104370432] Traceback (most recent call last):
[Mon Jul 29 13:45:39.941693 2019] [wsgi:error] [pid 17797:tid 139631104370432]   File "/usr/lib64/python3.6/threading.py", line 916, in _bootstrap_inner
[Mon Jul 29 13:45:39.941695 2019] [wsgi:error] [pid 17797:tid 139631104370432]     self.run()
[Mon Jul 29 13:45:39.941698 2019] [wsgi:error] [pid 17797:tid 139631104370432]   File "/usr/local/lib/python3.6/site-packages/flask_monitoringdashboard/core/profiler/stacktraceProfiler.py", line 77, in run
[Mon Jul 29 13:45:39.941700 2019] [wsgi:error] [pid 17797:tid 139631104370432]     self._on_thread_stopped()
[Mon Jul 29 13:45:39.941702 2019] [wsgi:error] [pid 17797:tid 139631104370432]   File "/usr/local/lib/python3.6/site-packages/flask_monitoringdashboard/core/profiler/stacktraceProfiler.py", line 93, in _on_thread_stopped
[Mon Jul 29 13:45:39.941705 2019] [wsgi:error] [pid 17797:tid 139631104370432]     self._outlier_profiler.add_outlier(request_id)
[Mon Jul 29 13:45:39.941707 2019] [wsgi:error] [pid 17797:tid 139631104370432] UnboundLocalError: local variable 'request_id' referenced before assignment


```

**Comment by bogdanp05** [comment]
Hi @bkb3! Thanks for signalling the error. It's a race condition that appears sometimes when monitoring level 3 is used. We are aware of it and a fix is already implemented on the `overhead` branch. 
We will release a new version, containing the fix, as soon as possible. In the meantime, you could use monitoring level 2 to avoid the error. It still contains the profiler, but not the outlier collection feature.

**Comment by mircealungu** [comment]
@bogdanp05 - is this fix already in the master? can we close the ticket?

**Comment by bogdanp05** [comment]
Hi @mircealungu ! The fix is in `development` but not yet in `master`. We should do a new release soon. Then we can close this issue and a few others.

**Comment by mircealungu** [comment]
Nice! We should release soon indeed!

**Comment by bogdanp05** [comment]
Fixed in #259
